pkg/api/folder.go

```diff
@@ -802,6 +804,19 @@ func (fk8s *folderK8sHandler) newToFolderDto(c *contextmodel.ReqContext, item un
 		return dtos.Folder{}, err
 	}
 
+	toID := func(rawIdentifier string) int64 {
+		parts := strings.Split(rawIdentifier, ":")
+		if len(parts) < 2 {
+			return 0
+		}
```

Comment:
Should this be an error ? Is ID=0 potentially the wrong user ?

Reply:
Thanks for the review! Based on the Slack convo this looks like it would be the anonymous user but I think it would be helpful to return an error. Made the change.